docs/source/supported_models.md

```diff
@@ -0,0 +1,39 @@
+# Supported Models and Hardware
```

Comment:
It'd be nice to have a brief sentence or two here about this section instead of leaving it empty. Maybe something like:

"TGI enables serving optimized models on specific hardware for the highest performance. The following sections list which models are hardware are supported."